frontend/pages/SoftwarePage/SoftwareAddPage/SoftwareCustomPackage/SoftwareCustomPackage.tsx

```diff
@@ -74,25 +91,37 @@ const SoftwareCustomPackage = ({
 
   const onSubmit = async (formData: IPackageFormData) => {
     console.log("submit", formData);
-    setIsUploading(true);
+    if (!formData.software) {
+      renderFlash(
+        "error",
+        `Couldn't add. Please refresh the page and try again.`
+      );
+      return;
+    }
 
     if (formData.software && formData.software.size > MAX_FILE_SIZE_BYTES) {
       renderFlash(
         "error",
         `Couldn't add. The maximum file size is ${MAX_FILE_SIZE_MB} MB.`
       );
-      setIsUploading(false);
       return;
     }
 
+    setUploadDetails(getFileDetails(formData.software));
+
     // Note: This TODO is copied to onSaveSoftwareChanges in EditSoftwareModal
     // TODO: confirm we are deleting the second sentence (not modifying it) for non-self-service installers
     try {
-      await softwareAPI.addSoftwarePackage(
-        formData,
-        currentTeamId,
-        UPLOAD_TIMEOUT
-      );
+      abortRef.current = new AbortController();
```

Comment:
why is this needed here if its don't on in the initial render useEffect?

Reply:
See the comment above. I think it's sort of analogous to the way you would set a new timeout for each request.